### PR TITLE
Backend: Django Permissions with Access Policy

### LINKED ISSUE #105: Backend: Add Permissions to all Models
We need to add permissions to only allow certain people to be able to access a specific API endpoint, etc.

### PR DESCRIPTION
**Pull request by aggarwal-h**
Closes #105.

### CODE REVIEW

**Comment by aggarwal-h** [comment]
@tevisdoe This is a big PR with lots of changes involving different permissions. I have set the permissions (access policies) with what seems reasonable, but if you feel that a group of users should/should not have access to something they do, please let me know.

Some things to point out:
- I have restricted all users from being able to make any PUT requests (but they are allowed to make PATCH updates). This was because I was running into issues with setting up serializers with certain properties.
- Some models (smaller models mainly) may be missing permissions still, but all the major ones are complete.
- I have used a library called [drf-access-policy](https://rsinger86.github.io/drf-access-policy/) to manage our permissions.
- When testing this code, I would advise that you go to `config > settings.py`, and edit `SIMPLE_JWT` settings as follows:
```
SIMPLE_JWT = {
    "AUTH_HEADER_TYPES": ("JWT",),
    "ACCESS_TOKEN_LIFETIME": timedelta(days=10),  # temporary
}
```
This will keep you from getting logged out while testing.

- You should test using the API `docs` that you created. You will need to click on the lock symbol on the page to add your access token manually. Remember to prefix your token with "JWT ". I would advise creating _ShopOwner_, _Employee_, and _Customer_ profiles separately, and then test the APIs with these different users. 

Let me know if you have any other questions/concerns.

File: src/django-app/quotes/policies.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        elif request.user.type == "shop_owner":
+            return qs.filter(shop__shop_owner=request.user)
+        elif request.user.type == "employee":
+            employee_shop = EmployeeData.objects.get(user=request.user).shop
+            return qs.filter(shop=employee_shop)
+        return qs
+
+    @classmethod
+    def scope_fields(cls, request, fields: dict, instance=None):
+        # TODO: This breaks the docs! Will leave this out.
```
Comment (tevisdoe) [inline src/django-app/quotes/policies.py:51]: If you meant to leave this in, that's fine, but you've commented to leave this bit of the code out, so I'm pointing it out incase you forgot to remove it.

**Review by tevisdoe — APPROVED** [review]
Gave this a whirl with a bunch of different combos of user and endpoints and it seems to work as intended, as far as I can tell. The permissions you've chosen for each user type seem logical enough, so I'm approving this. See my below comment before you merge.

File: src/django-app/quotes/policies.py (hunk 1 above)
Comment (aggarwal-h) [reply src/django-app/quotes/policies.py:51]: Yeah, my comment is a bit misleading. I have left out the feature (it is not scoping out fields currently) but I have left in the code for it so I can fix it in the future and implement it.

**Pull request merged**